lib/service/service.go

```diff
@@ -967,8 +967,14 @@ func NewTeleport(cfg *servicecfg.Config) (*TeleportProcess, error) {
 	imClient := cfg.InstanceMetadataClient
 	if imClient == nil {
 		imClient, err = cloud.DiscoverInstanceMetadata(supervisor.ExitContext())
-		if err != nil && !trace.IsNotFound(err) {
-			return nil, trace.Wrap(err)
+		if err == nil {
+			cfg.Logger.InfoContext(supervisor.ExitContext(),
+				"Found an instance metadata service. Teleport will import labels from this cloud instance",
```

Comment:
Suggestion: Since you have 2 sentences here I would keep the full stop, I think it's fine in cases like this.

```suggestion
				"Found an instance metadata service. Teleport will import labels from this cloud instance.",
```